magicctapipe/scripts/lst1_magic/lst_m1_m2_cloud_correction.py

```diff
@@ -0,0 +1,349 @@
+"""
+import argparse
+import logging
+
+import astropy.units as u
+import numpy as np
+import pandas as pd
+from astropy.coordinates import AltAz, SkyCoord
+from ctapipe.coordinates import TelescopeFrame
+from ctapipe.image import (
+    concentration_parameters,
+    hillas_parameters,
+    leakage_parameters,
+    timing_parameters,
+)
+from ctapipe.instrument import SubarrayDescription
+from ctapipe.io import read_table
+
+from magicctapipe.io import save_pandas_data_in_table
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.StreamHandler())
+logger.setLevel(logging.INFO)
+
+
+def model0(imp, h, zd):
+    """
+    Calculated the geometrical part of the model relating the emission height with the angular distance from the arrival direction
+
+    Parameters
+    ----------
+    imp : astropy.units.quantity.Quantity
+        Impact in m
+    h : astropy.units.quantity.Quantity
+        Height of each cloud layer a.g.l.
+    zd : numpy.float64
+        Zenith distance in deg
+
+    Returns
+    -------
+    numpy ndarray
+        Angular distance in units of degree
+    """
+    d = h / np.cos(zd)
+    return np.arctan((imp / d).to("")).to_value("deg")
+
+
+def model2(imp, h, zd):
+    """
+    Calculates the phenomenological correction to the distances obtained with model0
+
+    Parameters
+    ----------
+    imp : astropy.units.quantity.Quantity
+        Impact in m
+    h : astropy.units.quantity.Quantity
+        Height of each cloud layer a.g.l.
+    zd : numpy.float64
+        Zenith distance in deg
+
+    Returns
+    -------
+    astropy.units.quantity.Quantity
+        Angular distance corrected for bias in units of degrees
+    """
+    H0 = 2.2e3 * u.m
+    bias = 0.877 + 0.015 * ((h + H0) / (7.0e3 * u.m))
+    return bias * model0(imp, h, zd)
+
+
+def trans_height(x, Hc, dHc, trans):
+    """
+    Calculates transmission from a geometrically broad cloud at a set of heights.
+    Cloud is assumed to be homegeneous.
+
+    Parameters
+    ----------
+    x : astropy.units.quantity.Quantity
+        Height of each cloud layer a.g.l. in m
+    Hc : astropy.units.quantity.Quantity
+        Height of the base of the cloud a.g.l. in m
+    dHc : astropy.units.quantity.Quantity
+        Cloud thickness in m
+    trans : numpy.float64
+        Transmission of the cloud
+
+    Returns
+    -------
+    numpy.ndarray
+        Cloud transmission
+    """
+    t = pow(trans, ((x - Hc) / dHc).to_value(""))
+    t = np.where(x < Hc, 1, t)
+    t = np.where(x > Hc + dHc, trans, t)
+    return t
+
+
+def process_telescope_data(input_file, config, telid, focal_eff, camgeom):
+    """
+    Corrects LST-1 and MAGIC data affected by a cloud presence
+
+    Parameters
+    ----------
+    input_file : str
+        Path to an input .h5 DL1 file
+    config : dict
+        Configuration for the LST-1 + MAGIC analysis
+    telid : numpy.int16
+        LST-1 and MAGIC telescope ids
+    focal_eff : dict
+        Effective focal length
```

Comment:
since the function runs over a single tel_id at the time, why not pass directly the focal length of this telescope instead of the whole dictionary?